test-framework/arquillian/src/main/java/io/quarkus/arquillian/CreationalContextDestroyer.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.arquillian;
+
+import javax.enterprise.context.spi.CreationalContext;
+
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.core.spi.EventContext;
+import org.jboss.arquillian.test.spi.event.suite.After;
+
+public class CreationalContextDestroyer {
```

Comment:
Quarkus already has one Arq. observer registered - `QuarkusBeforeAfterLifecycle`.
We could just add the logic there.

Reply:
But we might want to rename that class, now that I think of it.
So it's just a matter of preference I think :)

Reply:
I don't know, it feels that they are doing quite different things, so the split into two classes makes sense to me :)